test/sim/moves/roost.js

```diff
@@ -78,6 +78,16 @@ describe('Roost', function () {
 		battle.makeChoices('move roost', 'move astonish');
 		assert.equal(battle.p1.active[0].hp, battle.p1.active[0].maxhp); // Immune to Astonish
 	});
+
+	it('should not remove Flying type during Terastallization', function () {
+		battle = common.createBattle([[
+			{species: "Dudunsparce", ability: "runaway", moves: ['roost'], teraType: "Flying"},
+		], [
+			{species: "Bonsly", ability: "sturdy", moves: ['earthquake']},
+		]]);
+		battle.makeChoices('move roost terastallize', 'auto');
```

Comment:
Roost won't remove the user's Flying-type at 100% HP either, so this test needs modified.